OptimizelySDK.Tests/OptimizelyTest.cs

```diff
@@ -1109,5 +1120,350 @@ public void TestGetVariationBucketingIdAttribute()
+            optly.SetFieldOrProperty("Config", tempConfig);
+
+            // Set such an experiment to the list of experiment ids, that does not belong to the feature.
+            featureFlag.ExperimentIds = new List<string> { "4209211" };
+
+            // Should return false when the experiment in feature flag does not get found in the datafile.
+            Assert.False((bool)optly.Invoke("IsFeatureEnabled", "multi_variate_feature", TestUserId, null));
+
+            LoggerMock.Verify(l => l.Log(LogLevel.ERROR, @"Experiment ID ""4209211"" is not in datafile."));
+        }
+
+        // Should return default value and log message when feature is not enabled for the user.
+        [Test]
+        public void TestIsFeatureEnabledGivenFeatureFlagIsNotEnabledForUser()
+        {
+            var featureKey = "double_single_variable_feature";
+            var featureFlag = Config.GetFeatureFlagFromKey("double_single_variable_feature");
+            
+            DecisionServiceMock.Setup(ds => ds.GetVariationForFeature(featureFlag, TestUserId, null)).Returns<Variation>(null);
+
+            var optly = Helper.CreatePrivateOptimizely();
+            optly.SetFieldOrProperty("DecisionService", DecisionServiceMock.Object);
+
+            bool result = (bool)optly.Invoke("IsFeatureEnabled", featureKey, TestUserId, null);
+            Assert.False(result);
+
+            LoggerMock.Verify(l => l.Log(LogLevel.INFO,
+                $@"Feature flag ""{featureKey}"" is not enabled for user ""{TestUserId}""."));
+        }
+
+        // Should return true when feature is enabled for the user but user does not get experimented.
+        [Test]
+        public void TestIsFeatureEnabledGivenFeatureFlagIsEnabledAndUserIsNotBeingExperimented()
+        {
+            var featureKey = "boolean_single_variable_feature";
+            var rollout = Config.GetRolloutFromId("166660");
+            var experiment = rollout.Experiments[0];
+            var variation = experiment.Variations[0];
+            var featureFlag = Config.GetFeatureFlagFromKey(featureKey);
+
+            DecisionServiceMock.Setup(ds => ds.GetVariationForFeature(featureFlag, TestUserId, null)).Returns(variation);
+
+            var optly = Helper.CreatePrivateOptimizely();
+            optly.SetFieldOrProperty("DecisionService", DecisionServiceMock.Object);
+
+            bool result = (bool)optly.Invoke("IsFeatureEnabled", featureKey, TestUserId, null);
+            Assert.True(result);
+
+            // SendImpressionEvent() does not get called.
+            LoggerMock.Verify(l => l.Log(LogLevel.INFO,
+                $@"The user ""{TestUserId}"" is not being experimented on feature ""{featureKey}""."), Times.Once);
+
+            LoggerMock.Verify(l => l.Log(LogLevel.INFO,
+                $@"Feature flag ""{featureKey}"" is enabled for user ""{TestUserId}""."));
+        }
+
+        // Should return true when feature is enabled for the user and user is being experimented.
```

Comment:
Should return true and send an impression event when feature is enabled for the user and user is being experimented

Reply:
Done